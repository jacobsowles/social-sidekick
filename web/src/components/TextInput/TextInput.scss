.text-input {
  border: 1px solid rgba(0, 0, 0, 0.1);
  margin-bottom: 10px;
  padding: 10px;
}
