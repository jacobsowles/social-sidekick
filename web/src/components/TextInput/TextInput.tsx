import classNames from 'classnames';
import React, { FunctionComponent } from 'react';

import './TextInput.scss';

export interface TextInputProps {
  className?: string;
  placeholder: string;
  value: string;
}

const TextInput: FunctionComponent<TextInputProps> = ({ className, placeholder, value }) => {
  return (
    <input
      aria-label={placeholder}
      className={classNames('text-input', className)}
      placeholder={placeholder}
      value={value}
    />
  );
};

export default TextInput;
