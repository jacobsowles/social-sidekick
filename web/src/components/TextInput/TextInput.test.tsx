import React from 'react';

import { init, shallow, toJson } from '@tests/component-test-base';
import TextInput, { TextInputProps } from './TextInput';

describe('TextInput', () => {
  let props: TextInputProps;
  init(() => <TextInput {...props} />);

  beforeEach(() => {
    props = {
      className: undefined,
      onChange: jest.fn(),
      placeholder: undefined,
      value: undefined
    };
  });

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });
});
