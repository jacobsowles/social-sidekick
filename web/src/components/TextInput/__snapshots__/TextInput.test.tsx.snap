// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render the component without crashing 1`] = `
<input
  className="text-input"
  onChange={[MockFunction]}
/>
`;
